### PR TITLE
Test openstack idp provider

### PR DESCRIPTION
**Pull request by ANKerD**

### CODE REVIEW

File: src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    private Map<String, String> createCredentials(String username, String password, String domain) {
+        Map<String, String> credentials = new HashMap<String, String>();
+
+        credentials.put(OpenStackConstants.Identity.USER_NAME_KEY_JSON, username);
+        credentials.put(OpenStackConstants.Identity.PASSWORD_KEY_JSON, password);
+        credentials.put(OpenStackConstants.Identity.DOMAIN_KEY_JSON, domain);
+
+        return credentials;
+    }
+
```
Comment (thiagomanel) [inline src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java]: is it possible to use this 4 parameters method and remove the above. In this case, pass null objects when testing the cases of missing parameters

File: src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        // exercise
+        this.plugin.getCloudUser(credentials);
+
+        // verify
+        Mockito.verify(this.plugin).getUnscopedCloudUserFromJson(Mockito.any());
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // with correct data, providing projectId
```
Comment (thiagomanel) [inline src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java]: this description misses the expected condition. Note that, it requires this pattern "When X then Y"

File: src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    public void testGetCloudUserWithoutProject() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // with correct data, without providing projectId
```
Comment (thiagomanel) [inline src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java]: this descriptions misses the expected condition. Note that, it requires this pattern "When X then Y"

File: src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java (hunk 1 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java]: Done.

File: src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java (hunk 3 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java]: Done.

File: src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java (hunk 2 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java]: Done.

**Pull request merged**